pkg/detectors/aws/aws.go

```diff
@@ -126,16 +155,28 @@ func (s scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 			}
 			resSecretMatch := strings.TrimSpace(secretMatch[1])
 
+			rawV2 := resIDMatch + resSecretMatch
+
 			s1 := detectors.Result{
 				DetectorType: detectorspb.DetectorType_AWS,
 				Raw:          []byte(resIDMatch),
 				Redacted:     resIDMatch,
-				RawV2:        []byte(resIDMatch + resSecretMatch),
+				RawV2:        []byte(rawV2),
 				ExtraData: map[string]string{
 					"resource_type": resourceTypes[idMatch[2]],
 				},
 			}
 
+			if val, ok := s.credsCache.Get(rawV2); ok {
+				item, ok := val.(cacheItem)
+				if !ok {
+					continue
```

Comment:
Not sure if this is the right behavior, i'd like to maybe log this as an error? Not sure if constructing our custom `logcontext.AddLogger` make sense so we can do that?